slsim/Sources/supernovae_host_match.py

```diff
@@ -0,0 +1,88 @@
+import numpy as np
+from astropy.table import Table
+import random
+
+"""References:
+Sullivan et al. 2006
+"""
+
+
+class SupernovaeHostMatch:
+    """Class to generate a host galaxy catalog for a given supernovae catalog."""
+
+    def __init__(
+        self,
+        supernovae_catalog,
+        galaxy_catalog,
+    ):
+        """
+
+        :param supernovae_catalog: supernovae redshift catalog
+        :type supernovae_catalog: np.ndarray
+        :param galaxy_catalog: supernovae host galaxy candidate catalog
+        :type galaxy_catalog: astropy Table
+        """
+        self.supernovae_catalog = supernovae_catalog
+        self.galaxy_catalog = galaxy_catalog
+
+    def match(self):
+        """Generates catalog in which supernovae are matched with host galaxies. (Fig 8
+        - Sullivan et al. 2006)
+
+        :return: catalog with supernovae redshifts and their corresponding host galaxies
+        :return type: dict
+        """
+        matched_catalog = Table(
+            names=(
+                "z",
+                "M",
+                "coeff",
+                "ellipticity",
+                "physical_size",
+                "stellar_mass",
+                "angular_size",
+                "mag_g",
+                "mag_r",
+                "mag_i",
+                "mag_z",
+                "mag_Y",
+            ),
+            dtype=(
+                "float64",
+                "float64",
+                "object",
+                "float64",
+                "float64",
+                "float64",
+                "float64",
+                "float64",
+                "float64",
+                "float64",
+                "float64",
+                "float64",
+            ),
+        )
+
+        # Iterate through the redshifts in the SNe catalog.
+        for redshift in self.supernovae_catalog:
+
+            # Select host galaxy candidates in the specified redshift range.
+            host_galaxy_candidates = self.galaxy_catalog[
+                (self.galaxy_catalog["z"] >= (redshift - 0.1))
```

Comment:
It mostly depends on the sky area used for the candidate galaxy catalog. With sky area 0.01 deg2, +/- 0.1 redshift range allows for ~100 host candidates to be considered in the stellar mass weighting. I will take this into account so that the range varies according to the sky area used for galaxy catalog.